vpc/service/internal_apis.go

```diff
@@ -407,3 +410,234 @@ func (vpcService *vpcService) doDetachBranchNetworkInterface(ctx context.Context
 		AssociationID: associationID,
 	}, nil
 }
+
+func (vpcService *vpcService) ResetSecurityGroup(ctx context.Context, in *vpcapi.SecurityGroupRequest) (*empty.Empty, error) {
+	var resetSgTx *sql.Tx
+	var err error
+	var count int
+
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+	ctx, span := trace.StartSpan(ctx, "ResetSecurityGroup")
+	defer span.End()
+
+	resetSgTx, err = beginSerializableTx(ctx, vpcService.db)
+	if err != nil {
+		err = errors.Wrap(err, "Unable to begin serializable transaction")
+		tracehelpers.SetStatus(err, span)
+		return &empty.Empty{}, err
+	}
+	defer func(tx *sql.Tx) {
+		_ = tx.Rollback()
+	}(resetSgTx)
+
+	sgToDelete := in.GetSgId()
+
+	logger.G(ctx).Debug("1. Check if the SG to be reset is associated with a container ", sgToDelete)
+	// 1. Check if the SG to be reset is associated with a container
+	rows, err := resetSgTx.QueryContext(ctx, `
+SELECT COUNT(*) FROM branch_enis
+WHERE ARRAY[$1] <@ security_groups AND branch_eni IN
+(SELECT branch_eni FROM branch_eni_attachments WHERE branch_eni_attachments.association_id IN
```

Comment:
I think this query should actually be:

```sql

SELECT branch_eni,
       account_id,
       az,
  (SELECT count(*)
   FROM assignments
   WHERE branch_eni_association =
       (SELECT association_id
        FROM branch_eni_attachments bea
        WHERE bea.branch_eni = branch_enis.branch_eni
          AND state = 'attached'))
FROM branch_enis
WHERE security_groups @> $1::text[]
```
This gives you the AZ, and branch_eni IDs. You then check the count column to make sure it's 0.
You can then give the user feedback to which specific branch_enis have a count of > 0 as well.

Reply:
you could join the availability_zones table in to get the region as well.